packages/clay-tooltip/src/TooltipProvider.tsx

```diff
@@ -117,18 +133,49 @@ type TContentRenderer = (props: {
 	title: string;
 }) => React.ReactElement | React.ReactNode;
 
-const TooltipProvider: React.FunctionComponent<{
+interface IPropsBase {
+	/**
+	 * Flag to indicate if tooltip should automatically align based on the window
+	 */
 	autoAlign?: boolean;
-	children: React.ReactElement;
+
+	/**
+	 * Custom function for rendering the contents of the tooltip
+	 */
 	contentRenderer?: TContentRenderer;
+
+	/**
+	 * Delay in miliseconds before showing tooltip
+	 */
 	delay?: number;
-}> = ({
+}
+
+interface IPropsWithChildren extends IPropsBase {
+	children: React.ReactElement;
+	scope?: never;
+}
+
+interface IPropsWithScope extends IPropsBase {
+	children?: never;
+
+	/**
+	 * CSS selector to scope provider to. All titles within this scope will be
+	 * rendered in the tooltip. Titles outside of this scope will be styled
+	 * as with the default browser.
+	 */
+	scope: string;
+}
+
+const TooltipProvider: React.FunctionComponent<
+	IPropsWithChildren | IPropsWithScope
```

Comment:
This is interesting. Can TypeScript alert you if you pass both props? that would be pretty cool.

Reply:
It looks like it an infer that if `scope` exists then children will be undefined.

[test playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgEZwM4oN4CgCQA9iBAPwBcyGYUoA5rgL666iSyIoyGHIQAekEABMMaTDgJgA7oUrVaIBvjAALKBDKUSAN2hMWbaPCTiofQRBFj0WZHhWyKyXfpXrN8mvQO4ExamQAFQhqSgAlCEQwADoAMQBXEAQwYGIAYUIAWwAHYiswAB5uXgAfMwA+ZABeZAAKHKhCHIwAShqqh2AYesbmjBiZQnaHfH8QQJ04ABsa5D6WwY8IAjGAsGQp6YAmOYWBoYJmAg0wBKgQZELhYB1kAHoKpiA)